Gems/ScriptCanvas/Code/Editor/SystemComponent.cpp

```diff
@@ -197,6 +199,68 @@ namespace ScriptCanvasEditor
         return AzToolsFramework::AssetBrowser::SourceFileDetails();
     }
 
+    void SystemComponent::AddSourceFileCreators
+        ( [[maybe_unused]] const char* fullSourceFolderName
+        , [[maybe_unused]] const AZ::Uuid& sourceUUID
+        , AzToolsFramework::AssetBrowser::SourceFileCreatorList& creators)
+    {
+        auto scriptCavnasAssetCreator = [](const AZStd::string& fullSourceFolderNameInCallback, [[maybe_unused]] const AZ::Uuid& sourceUUID)
+        {
+            const AZStd::string defaultFilename = "NewScript";
+            const AZStd::string scriptCanvasExtension = ScriptCanvasEditor::SourceDescription::GetFileExtension();
+
+            AZStd::string fullFilepath;
+            AZ::StringFunc::Path::ConstructFull(fullSourceFolderNameInCallback.c_str()
+                , defaultFilename.c_str()
+                , scriptCanvasExtension.c_str()
+                , fullFilepath);
+
+            int fileCounter = 0;
+            while (AZ::IO::FileIOBase::GetInstance()->Exists(fullFilepath.c_str()))
+            {
+                fileCounter++;
+                const AZStd::string incrementalFilename = defaultFilename + AZStd::to_string(fileCounter);
+
+                AZ::StringFunc::Path::ConstructFull(fullSourceFolderNameInCallback.c_str()
+                    , incrementalFilename.c_str()
+                    , scriptCanvasExtension.c_str()
+                    , fullFilepath);
+            }
+
+            const AZ::IO::Path fullAzFilePath = fullFilepath;
+            const ScriptCanvas::DataPtr graph = EditorGraph::Create();
+            const AZ::Uuid assetId = AZ::Uuid::CreateRandom();
```

Comment:
Don't create a random guid. The guid should be handled by other systems if necessary.

Reply:
Done. Retested to make sure it still creates a valid SC asset.